solutions/poi-18-BikePaths.mdx

```diff
@@ -0,0 +1,99 @@
+---
+id: poi-18-BikePaths
+source: POI
+title: 2018 - Bike paths
+author: Neo Wang
+---
+
+## Explanation
+
+Notice that groups of intersections can **form strongly connected components.**
+Therefore, treat each component as a node with initial value equal to the size
+of the component. Since the resulting strongly connected components
+$\texttt{comp}$ form a directed acyclic graph (specifically a tree, because of the "justness" from the problem statement), process the edges $e$ in reverse
```

Comment:
Perhaps make it clear that this solution *only* works because it's a tree (and explain why that is) -- So that people don't go around thinking this works on a general dag

Reply:
could you elaborate (I tried various dags, but they all seem to work j fine), though I am probably missing something

Reply:
you can't just do `dp[i] += dp[child]` to count how many nodes `i` can reach for a DAG (except for trees).

example: (`a->b`,`a->c`,`b->d`,`c->d`, a DAG but not a tree)
- `dp[d] = 1`
- `dp[b] = 1 + dp[d] = 2`
- `dp[c] = 1 + dp[d] + 2`
- `dp[a] = 1 + dp[b] + dp[c] = 5` (not correct, `d` is counted twice)